controller/src/main/java/io/pravega/controller/server/rest/resources/HealthImpl.java

```diff
@@ -0,0 +1,292 @@
+                    asyncResponse,
+                    method,
+                    traceId,
+                    contributorNotFound(id));
+        } catch (RuntimeException e) {
+            asyncResponse.resume(Response.status(Response.Status.INTERNAL_SERVER_ERROR).build());
+        } finally {
+            LoggerHelpers.traceLeave(log, method, traceId);
+        }
+    }
+
+    @Override
+    public void getContributorDetails(String id, SecurityContext securityContext, AsyncResponse asyncResponse) throws NotFoundException {
+        getDetails(id, securityContext, asyncResponse, "getDetails");
+    }
+
+    @Override
+    public void getDetails(SecurityContext securityContext, AsyncResponse asyncResponse) throws NotFoundException {
+        getDetails(null, securityContext, asyncResponse, "getDetails");
+    }
+
+    private void getDetails(String id, SecurityContext securityContext, AsyncResponse asyncResponse, String method) {
+        long traceId = LoggerHelpers.traceEnter(log, method);
+        try {
+            restAuthHelper.authenticateAuthorize(getAuthorizationHeader(), authorizationResource.ofScopes(), READ_UPDATE);
+            Map<String, Object> details = endpoint.getDetails(id);
+            asyncResponse.resume(Response.status(Response.Status.OK)
+                    .entity(adapter(details))
+                    .build());
+        } catch (AuthException e) {
+            processException(e.getResponseCode(),
+                    asyncResponse,
+                    method,
+                    traceId,
+                    authenticationFailure(id, "Details"));
+        } catch (ContributorNotFoundException e) {
+            processException(e.getResponseCode(),
+                    asyncResponse,
+                    method,
+                    traceId,
+                    contributorNotFound(id));
+        } finally {
+            LoggerHelpers.traceLeave(log, method, traceId);
+        }
+    }
+
+    @Override
+    public void getContributorReadiness(String id, SecurityContext securityContext, AsyncResponse asyncResponse) throws NotFoundException {
+        getReadiness(id, securityContext, asyncResponse, "getContributorReadiness");
+    }
+
+    @Override
+    public void getReadiness(SecurityContext securityContext, AsyncResponse asyncResponse) throws NotFoundException {
+        getReadiness(null, securityContext, asyncResponse, "getReadiness");
+    }
+
+    private void getReadiness(String id, SecurityContext securityContext, AsyncResponse asyncResponse, String method) {
+        long traceId = LoggerHelpers.traceEnter(log, method);
+        try {
+            restAuthHelper.authenticateAuthorize(getAuthorizationHeader(), authorizationResource.ofScopes(), READ_UPDATE);
+            boolean ready = endpoint.isReady(id);
+            asyncResponse.resume(Response.status(Response.Status.OK)
+                    .entity(ready)
+                    .build());
+        } catch (AuthException e) {
```

Comment:
It would be nice if we could factor out this repeated try-catch logic into its own method. That should simplify the code in this class.

For example, 

```
private void handleException(Runnable r) {
  try {
    r.run();
  } /// put ugly catch code here
}
```

then you'd invoke it like:
```
handleException(() -> { your code here });
```